packages/sui-react-initial-props/src/createClientPageComponent.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+
+// This method creates the PageComponent prepared for the Client Side Rendering
+// It renders a placeholder, if any specified as `renderLoading` on the PageComponent
+// Also, executes the getInitialProps to retrieve the initialProps and then
+// Renders the component with the initialProps, hiding the placeholder
+export default ({ Page, context, routeInfo }) => (
```

Comment:
You rewrite this HoC in the way:

```
export default ({context, routeInfo}) => Page => class ClientPage extends Component {
  static displayName = `ClientPage(${Page.displayName})`
}
```

Doing that we can combine this HoC with others